lib/std/crypto/ecdsa.zig

```diff
@@ -26,7 +26,7 @@ pub const EcdsaSecp256k1Sha256oSha256 = Ecdsa(crypto.ecc.Secp256k1, crypto.hash.
 
 /// Elliptic Curve Digital Signature Algorithm (ECDSA).
 pub fn Ecdsa(comptime Curve: type, comptime Hash: type) type {
-    const Hmac = crypto.auth.hmac.Hmac(Hash);
+    const Hmac = if (@hasDecl(Hash, "ToMac")) Hash.ToMac() else crypto.auth.hmac.Hmac(Hash);
```

Comment:
#19352

suggest to create an explicit decl that advertises whether it supports being used with Hmac.

Reply:
HMAC can be instantiated with SHAKE, but that would be wasting CPU cycles.
Anyway, the function has been removed, and current SHAKE variants are now individually handled in `ecdsa.zig`.